test/backup_test.js

```diff
@@ -151,64 +157,342 @@ describe('Backup', function() {
+      const conversation = {
         active_at: 123,
         id: 'id',
-        type: 'private'
+        type: 'private',
       };
-      var expected = '123 (id)';
-      assert.strictEqual(Signal.Backup._getConversationLoggingName(conversation), expected);
+      const expected = '123 (id)';
+      assert.strictEqual(
+        Signal.Backup._getConversationLoggingName(conversation),
+        expected
+      );
     });
 
-    it('uses just id if name is not available', function() {
-      var conversation = {
+    it('uses just id if name is not available', () => {
+      const conversation = {
         active_at: 123,
         id: 'groupId',
-        type: 'group'
+        type: 'group',
       };
-      var expected = '123 ([REDACTED_GROUP]pId)';
-      assert.strictEqual(Signal.Backup._getConversationLoggingName(conversation), expected);
+      const expected = '123 ([REDACTED_GROUP]pId)';
+      assert.strictEqual(
+        Signal.Backup._getConversationLoggingName(conversation),
+        expected
+      );
     });
 
-    it('uses inactive for missing active_at', function() {
-      var conversation = {
+    it('uses inactive for missing active_at', () => {
+      const conversation = {
         id: 'id',
-        type: 'private'
+        type: 'private',
       };
-      var expected = 'inactive (id)';
-      assert.strictEqual(Signal.Backup._getConversationLoggingName(conversation), expected);
+      const expected = 'inactive (id)';
+      assert.strictEqual(
+        Signal.Backup._getConversationLoggingName(conversation),
+        expected
+      );
+    });
+  });
+
+  describe('end-to-end', () => {
+    it('exports then imports to produce the same data we started with', async () => {
+      const {
+        attachmentsPath,
+        fs,
+        fse,
+        glob,
+        path,
+        tmp,
+      } = window.test;
+      const {
+        upgradeMessageSchema,
+        loadAttachmentData,
+      } = window.Signal.Migrations;
+
+      const key = new Uint8Array([
+        1, 3, 4, 5, 6, 7, 8, 11,
+        23, 34, 1, 34, 3, 5, 45, 45,
+        1, 3, 4, 5, 6, 7, 8, 11,
+        23, 34, 1, 34, 3, 5, 45, 45,
+      ]);
+      const attachmentsPattern = path.join(attachmentsPath, '**');
+
+      const OUR_NUMBER = '+12025550000';
+      const CONTACT_ONE_NUMBER = '+12025550001';
+
+      async function wrappedLoadAttachment(attachment) {
+        return _.omit(await loadAttachmentData(attachment), ['path']);
+      }
+
+      async function clearAllData() {
+        await textsecure.storage.protocol.removeAllData();
+        await fse.emptyDir(attachmentsPath);
+      }
+
+      function removeId(model) {
+        return _.omit(model, ['id']);
+      }
+
+      // On windows, attachmentsPath has a normal windows path format (\ separators), but
+      //   glob returns only /. We normalize to / separators for our manipulations.
+      const twoSlashes = /[^/]*\/[^/]*\/[^/]*/;
```

Comment:
Can you clarify where two slashes come from and what we do with them? Maybe with an example…